pkg/logs/scheduler/scheduler.go

```diff
@@ -172,6 +174,15 @@ func (s *Scheduler) toSources(config integration.Config) ([]*logsConfig.LogSourc
 	}
 
 	var service *service.Service
+
+	commonGlobalOptions := integration.CommonGlobalConfig{}
+	err = yaml.Unmarshal(config.InitConfig, &commonGlobalOptions)
+	if err != nil {
+		return nil, fmt.Errorf("invalid init_config section for source %s: %s", config.Name, err)
+	}
+
+	globalServiceDefined := len(commonGlobalOptions.Service) > 0
```

Comment:
Because `commonGlobalOptions.Service` is a string, I think it would be better to have: 
```go

globalServiceDefined := commonGlobalOptions.Service != ""
```